### PR TITLE
Various improvements / nits

### PR DESCRIPTION
**Pull request by spolu**
More precise errors, added managed flag to plan.limits.dataSources, remove some redirects for better error reporting.

### CODE REVIEW

**Review by lasryaric — CHANGES_REQUESTED** [review]

File: front/lib/connectors_api.ts [hunk 1]
```diff
@@ -12,20 +12,20 @@ export type ConnectorsAPIResponse<T> = Result<T, ConnectorsAPIErrorResponse>;
 
 export const ConnectorsAPI = {
   async createConnector(
-    provider: string,
+    provider: "slack" | "notion",
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/connectors_api.ts:15]: We can't pull from `connectors` can we? removing `google` there.

File: front/pages/api/w/[wId]/data_sources/index.ts [hunk 2]
```diff
@@ -37,8 +51,14 @@ async function handler(
 
     case "POST":
       if (!auth.isBuilder()) {
-        res.status(401).end();
-        return;
+        return apiError(req, res, {
+          status_code: 401,
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/pages/api/w/[wId]/data_sources/index.ts:55]: Sure will update them all.

File: front/lib/connectors_api.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lasryaric) [reply front/lib/connectors_api.ts:15]: In short we can't. Connectors has too many heavy dependencies and they don't play nicely with front.

File: front/lib/auth.ts [hunk 3]
```diff
@@ -358,19 +370,24 @@ export function planForWorkspace(w: Workspace): PlanType {
...
 }
 
+/**
+ * Retrieves or create a system API key for a given workspace
+ * @param workspace WorkspaceType
+ * @returns Promise<Result<Key, Error>>
+ */
 export async function getOrCreateSystemApiKey(
-  workspaceId: number
+  workspace: WorkspaceType
```
[the remark this answers is not in the document]
Comment (lasryaric) [reply front/lib/auth.ts:379]: Agree for passing a type, not necessarily to pass the whole object but not a big deal.
We can have a type for Id that just overrides number:
```typescript
type ModelId = number;
```
I am fine with passing the workspace here. But that could be a decent solution for our entire code base.

**Review by lasryaric — APPROVED** [review]

**Comment by spolu** [comment]
> In short we can't. Connectors has too many heavy dependencies and they don't play nicely with front.

So I'm not sure I parse:

> Lets add and export a typed list of providers in this file, which should stay in sync with connectors/src/types/connector.ts

?

**Pull request merged**